### PR TITLE
Feat: Added Wishlist Functionality

### LINKED ISSUE #11: todo: add wishlist functionality

### PR DESCRIPTION
**Pull request by coldpigli**
Added Wishlist Functionality:
You can now:

1) Add to Wishlist
2) Remove From Wishlist
3) Proper checks to see if item is already in wishlist or not

### CODE REVIEW

File: src/components/ProductItem.jsx [hunk 1]
```diff
@@ -1,20 +1,39 @@
...
 
 const ProductItem = ({product}) => {
 
     const {imageUrl,isLiked,title,quantity,price,rating} = product;
+    const {isLoggedIn, userDetails,setUserDetails} = useAuth();
+    const {wishList,cartList} = userDetails;
+    const navigate = useNavigate();
 
     const checkClick=()=>{
         console.log("Add to Cart")
```
Comment (rahulcs754) [inline src/components/ProductItem.jsx:12]: don't leave console logs

File: src/components/ProductItem.jsx [hunk 2]
```diff
@@ -1,20 +1,39 @@
...
     }
 
+    const {allProducts, setAllProducts} = useProducts();
+
+    const addToWishlist = (product) => {
+        if(isLoggedIn){
+            (wishList.find((item)=>item._id===product._id))
+            ?console.log("Item already exists"):
```
Comment (rahulcs754) [inline src/components/ProductItem.jsx:20]: you can use any toast for say "item already exits"

File: src/components/WishlistItem.jsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+                <div className="cards-cta">
+                    <div className="quantity-counter flex">
+                    <h2 className="heading3">₹{price}</h2>
+                    </div>
+                    <div className="quantity-counter flex">
+                        {/* ToDo: Will toogle on the basis of count later  */}
+                        {/* <div className="counter-button bod-light heading3 children-middle">-</div>
+                        <p>2</p>
+                        <div className="counter-button bod-light heading3 children-middle">+</div> */}
```
Comment (rahulcs754) [inline src/components/WishlistItem.jsx:50]: remove comment code

**Review by rahulcs754 — APPROVED** [review]
the code is clean and good 
just few suggestions given by me please check

**Review by sadathshariff — APPROVED** [review]
Looks Good Clean code and good naming Conventions used

**Pull request merged**